diff --git a/libheif/plugins/decoder_vvdec.cc b/libheif/plugins/decoder_vvdec.cc
index 0951572..380404e 100644
--- a/libheif/plugins/decoder_vvdec.cc
+++ b/libheif/plugins/decoder_vvdec.cc
@@ -116,6 +116,7 @@ heif_error vvdec_new_decoder2(void** dec, const heif_decoder_plugin_options* opt
   if (options->num_threads) {
     params.threads = options->num_threads;
   }
+  params.threads = 0; // Single Thread
   decoder->decoder = vvdec_decoder_open(&params);
 
   const int MaxNaluSize = 256 * 1024;
diff --git a/libheif/plugins/encoder_vvenc.cc b/libheif/plugins/encoder_vvenc.cc
index 5f10940..f9ea54d 100644
--- a/libheif/plugins/encoder_vvenc.cc
+++ b/libheif/plugins/encoder_vvenc.cc
@@ -462,6 +462,7 @@ static heif_error vvenc_start_sequence_encoding_intern(void* encoder_raw, const
                                0,
                                encoder_quality,
                                VVENC_MEDIUM);
+  params.m_numThreads = 0; // Single Thread
   if (ret != VVENC_OK) {
     // TODO: cleanup memory
 
