diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8ff0a6e..ab18928 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -64,22 +64,22 @@ endif()
 if( ${CMAKE_SYSTEM_NAME} STREQUAL "Emscripten" )
   set( VVDEC_TARGET_WASM TRUE )
 
-  add_compile_options( -pthread )
-  add_link_options(
-    --bind
-    -sWASM_BIGINT
-    -sINITIAL_MEMORY=1300MB # this should be enough for FullHD decoding
-    -sINVOKE_RUN=0          # don't call main() automatically
-    -sUSE_PTHREADS
-    #-sPROXY_TO_PTHREAD
-    #-sPTHREAD_POOL_SIZE=20
-    -sMINIFY_HTML=0
-    -sMODULARIZE
-    -sEXPORT_NAME=CreateVVdeC
-    -sEXPORTED_RUNTIME_METHODS=[ccall,cwrap,getValue,setValue,_malloc,callMain,FS]
-    -sEXPORTED_FUNCTIONS=@${CMAKE_CURRENT_SOURCE_DIR}/source/Lib/vvdec/wasm_exported_functions.json
-#    --preload-file=${CMAKE_CURRENT_SOURCE_DIR}/ext/bitstreams/AMVR_B_HHI_3/AMVR_B_HHI_3.bit@in.bit
-  )
+#   add_compile_options( -pthread )
+#   add_link_options(
+#     --bind
+#     -sWASM_BIGINT
+#     -sINITIAL_MEMORY=1300MB # this should be enough for FullHD decoding
+#     -sINVOKE_RUN=0          # don't call main() automatically
+#     -sUSE_PTHREADS
+#     #-sPROXY_TO_PTHREAD
+#     #-sPTHREAD_POOL_SIZE=20
+#     -sMINIFY_HTML=0
+#     -sMODULARIZE
+#     -sEXPORT_NAME=CreateVVdeC
+#     -sEXPORTED_RUNTIME_METHODS=[ccall,cwrap,getValue,setValue,_malloc,callMain,FS]
+#     -sEXPORTED_FUNCTIONS=@${CMAKE_CURRENT_SOURCE_DIR}/source/Lib/vvdec/wasm_exported_functions.json
+# #    --preload-file=${CMAKE_CURRENT_SOURCE_DIR}/ext/bitstreams/AMVR_B_HHI_3/AMVR_B_HHI_3.bit@in.bit
+#   )
 
   if( VVDEC_ENABLE_X86_SIMD )
     add_compile_options( -msimd128 )            # needed for all SIMD code, but breaks running in nodejs (firefox & chrome do work)
